includes/class-convertkit-setup.php

```diff
@@ -104,6 +113,133 @@ public function update() {
 
 	}
 
+	/**
+	 * 2.5.3: Migrate Third Party Form integrations' 'None' option values from `default` to blank.
+	 *
+	 * 2.4.9 changed the 'None' label's value from `default` to a blank string, as the v4 API's
+	 * `add_subscriber_to_form()` method introduces type declarations, which would result in
+	 * an uncaught TypeError when passing a non integer value.
+	 *
+	 * The PR for that (https://github.com/ConvertKit/convertkit-wordpress/pull/655) didn't include
+	 * any tests or upgrade/migration routines to change any existing saved settings where the 'None'
+	 * label's value was stored as `default`.
+	 */
+	private function migrate_contact_form_7_none_setting() {
+
+		$convertkit_contact_form_7_settings = new ConvertKit_ContactForm7_Settings();
+
+		// Bail if no settings exist.
+		if ( ! $convertkit_contact_form_7_settings->has_settings() ) {
+			return;
+		}
+
+		// Get settings.
+		$settings = $convertkit_contact_form_7_settings->get();
+
+		// Iterate through settings.
+		foreach ( $settings as $contact_form_7_form_id => $convertkit_form_id ) {
+			// Skip keys that are non-numeric e.g. `creator_network_recommendations_*`.
+			if ( ! is_numeric( $contact_form_7_form_id ) ) {
+				continue;
+			}
+
+			// Skip values that are not 'default'.
+			if ( $convertkit_form_id !== 'default' ) {
+				continue;
+			}
+
+			// Change 'default' to a blank string.
+			$settings[ $contact_form_7_form_id ] = '';
+		}
+
+		// Update settings.
+		update_option( $convertkit_contact_form_7_settings::SETTINGS_NAME, $settings );
+
+	}
+
+	/**
+	 * 2.5.3: Migrate Third Party Form integrations' 'None' option values from `default` to blank.
+	 *
+	 * 2.4.9 changed the 'None' label's value from `default` to a blank string, as the v4 API's
+	 * `add_subscriber_to_form()` method introduces type declarations, which would result in
+	 * an uncaught TypeError when passing a non integer value.
+	 *
+	 * The PR for that (https://github.com/ConvertKit/convertkit-wordpress/pull/655) didn't include
+	 * any tests or upgrade/migration routines to change any existing saved settings where the 'None'
+	 * label's value was stored as `default`.
+	 */
+	private function migrate_forminator_none_setting() {
+
+		$convertkit_forminator_settings = new ConvertKit_Forminator_Settings();
+
+		// Bail if no settings exist.
+		if ( ! $convertkit_forminator_settings->has_settings() ) {
+			return;
+		}
+
+		// Get settings.
+		$settings = $convertkit_forminator_settings->get();
+
+		// Iterate through settings.
+		foreach ( $settings as $forminator_form_id => $convertkit_form_id ) {
+			// Skip keys that are non-numeric e.g. `creator_network_recommendations_*`.
+			if ( ! is_numeric( $forminator_form_id ) ) {
+				continue;
+			}
+
+			// Skip values that are not 'default'.
+			if ( $convertkit_form_id !== 'default' ) {
+				continue;
```

Comment:
Why not write it more directly?

```php
if ( $convertkit_form_id === 'default' ) {
    $settings[ $forminator_form_id ] = '';
}
```

Negatives are harder to process when reading code, and it saves a line!

Reply:
Most of the Plugins have been written in this coding style. Curious why this is now an issue?